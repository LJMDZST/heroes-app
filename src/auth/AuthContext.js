
import {createContext} from 'react';

/** Objeto Context que sirve para gestionar la informacion
 *  de la autentificacion de los usuarios.
 */

export const AuthContext = createContext();