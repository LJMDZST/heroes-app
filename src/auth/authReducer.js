import {types} from '../types/types.js';

/** Reducer que modifica el estado del usuario
 *  de acuerdo a la accion recibida
 */

export const authReducer = (state = {}, action)=>{
    switch ( action.type ) {
        case types.login:
            return {
                ...action.payload,
                logged: true
            }
        case types.logout:
            return {
                logged: false
            }
    
        default:
            return state;
    }
}