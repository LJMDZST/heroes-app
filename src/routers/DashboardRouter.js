import React  from 'react';
import { Redirect, Route, Switch } from 'react-router';


import { NavBar } from '../components/ui/NavBar.js';
import { MarvelScreen } from "../components/marvel/MarvelScreen.js";
import { DcScreen } from '../components/dc/DcScreen.js';
import { HeroScreen } from '../components/heroes/HeroScreen.js';
import { SearchScreen } from '../components/search/SearchScreen.js';

/** Compoente que contiene la barra de navegacion y un 
 *  switch que redirecciona a cada una de las pantallas 
 *  de la app.
 *  DashBoard 
 *         -> /marvel (inicio)
 *         -> /heroe
 *         -> /dc
 */
export const DashboardRouter = ()  => {
    return (
        <>
            {/* 
                El NavBar se visualizara en todas 
                las pantallas que redireccione el switch
            */}
            <NavBar />
            <div className="container mt-2">
                <Switch >
                    <Route path="/marvel" component={ MarvelScreen } />
                    <Route path="/heroe/:heroeId" component={ HeroScreen } />
                    <Route path="/dc" component={ DcScreen } />
                    <Route path="/search" component={ SearchScreen } />

                    <Redirect to="/marvel" />
                </Switch>
            </div>
        </>
    )
}
