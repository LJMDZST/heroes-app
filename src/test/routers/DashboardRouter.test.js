import { mount }  from 'enzyme';
import { MemoryRouter } from 'react-router';
import { AuthContext } from '../../auth/AuthContext';
import { DashboardRouter } from '../../routers/DashboardRouter';


describe('Pruebas en <DashboardRouters />', () => {
    

    test('debe mostrarse correctamente', () => {
        const contextValue = {
            dispatch : jest.fn(),
            user : {
                    logged :false
                }
        }
        
        const wrapper = mount(
            <AuthContext.Provider value = {contextValue}>
                <MemoryRouter>
                    <DashboardRouter />
                </MemoryRouter>
            </AuthContext.Provider>
        );

        expect(wrapper).toMatchSnapshot();

    })
    

})
