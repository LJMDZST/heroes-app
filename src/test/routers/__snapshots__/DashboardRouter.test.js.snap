// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <DashboardRouters /> debe mostrarse correctamente 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          aria-current="page"
          className="nav-item nav-link active"
          href="/marvel"
          onClick={[Function]}
          style={Object {}}
        >
          Marvel
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2"
    >
      <ul
        className="navbar-nav ms-auto"
      >
        <span
          className="nav-item nav-link text-info"
        />
        <button
          className="nav-item nav-link btn"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container mt-2"
  >
    <div>
      <h1>
        Marvel Screen
      </h1>
      <hr />
      <div
        className="row row-cols-1 row-cols-md-3 g-3 animate__animated animate__fadeIn"
      >
        <div
          className="col"
        >
          <div
            className="card"
          >
            <div
              className="card-header"
            >
              <h5
                className="card-title fw-bolder"
              >
                Spider Man
              </h5>
            </div>
            <img
              alt="Spider Man"
              className="card-img-top"
              src="./assets/heroes/marvel-spider.jpg"
            />
            <div
              className="card-body "
            >
              <p
                className="card-text fw-bold"
              >
                Peter Parker
              </p>
              <p
                className="card-text"
              >
                Amazing Fantasy #15
              </p>
            </div>
            <div
              className="d-flex justify-content-between card-footer"
            >
              <small
                className="text-muted"
              >
                Marvel Comics
              </small>
              <a
                href="/heroe/marvel-spider"
                onClick={[Function]}
              >
                Más...
              </a>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
          >
            <div
              className="card-header"
            >
              <h5
                className="card-title fw-bolder"
              >
                Captain America
              </h5>
            </div>
            <img
              alt="Captain America"
              className="card-img-top"
              src="./assets/heroes/marvel-captain.jpg"
            />
            <div
              className="card-body "
            >
              <p
                className="card-text fw-bold"
              >
                Steve Rogers
              </p>
              <p
                className="card-text"
              >
                Captain America Comics #1
              </p>
            </div>
            <div
              className="d-flex justify-content-between card-footer"
            >
              <small
                className="text-muted"
              >
                Marvel Comics
              </small>
              <a
                href="/heroe/marvel-captain"
                onClick={[Function]}
              >
                Más...
              </a>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
          >
            <div
              className="card-header"
            >
              <h5
                className="card-title fw-bolder"
              >
                Iron Man
              </h5>
            </div>
            <img
              alt="Iron Man"
              className="card-img-top"
              src="./assets/heroes/marvel-iron.jpg"
            />
            <div
              className="card-body "
            >
              <p
                className="card-text fw-bold"
              >
                Tony Stark
              </p>
              <p
                className="card-text"
              >
                Tales of Suspense #39
              </p>
            </div>
            <div
              className="d-flex justify-content-between card-footer"
            >
              <small
                className="text-muted"
              >
                Marvel Comics
              </small>
              <a
                href="/heroe/marvel-iron"
                onClick={[Function]}
              >
                Más...
              </a>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
          >
            <div
              className="card-header"
            >
              <h5
                className="card-title fw-bolder"
              >
                Thor
              </h5>
            </div>
            <img
              alt="Thor"
              className="card-img-top"
              src="./assets/heroes/marvel-thor.jpg"
            />
            <div
              className="card-body "
            >
              <p
                className="card-text fw-bold"
              >
                Thor Odinson
              </p>
              <p
                className="card-text"
              >
                Journey into Myster #83
              </p>
            </div>
            <div
              className="d-flex justify-content-between card-footer"
            >
              <small
                className="text-muted"
              >
                Marvel Comics
              </small>
              <a
                href="/heroe/marvel-thor"
                onClick={[Function]}
              >
                Más...
              </a>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
          >
            <div
              className="card-header"
            >
              <h5
                className="card-title fw-bolder"
              >
                Hulk
              </h5>
            </div>
            <img
              alt="Hulk"
              className="card-img-top"
              src="./assets/heroes/marvel-hulk.jpg"
            />
            <div
              className="card-body "
            >
              <p
                className="card-text fw-bold"
              >
                Bruce Banner
              </p>
              <p
                className="card-text"
              >
                The Incredible Hulk #1
              </p>
            </div>
            <div
              className="d-flex justify-content-between card-footer"
            >
              <small
                className="text-muted"
              >
                Marvel Comics
              </small>
              <a
                href="/heroe/marvel-hulk"
                onClick={[Function]}
              >
                Más...
              </a>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
          >
            <div
              className="card-header"
            >
              <h5
                className="card-title fw-bolder"
              >
                Wolverine
              </h5>
            </div>
            <img
              alt="Wolverine"
              className="card-img-top"
              src="./assets/heroes/marvel-wolverine.jpg"
            />
            <div
              className="card-body "
            >
              <p
                className="card-text fw-bold"
              >
                James Howlett
              </p>
              <p
                className="card-text"
              >
                The Incredible Hulk #180
              </p>
            </div>
            <div
              className="d-flex justify-content-between card-footer"
            >
              <small
                className="text-muted"
              >
                Marvel Comics
              </small>
              <a
                href="/heroe/marvel-wolverine"
                onClick={[Function]}
              >
                Más...
              </a>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
          >
            <div
              className="card-header"
            >
              <h5
                className="card-title fw-bolder"
              >
                Daredevil
              </h5>
            </div>
            <img
              alt="Daredevil"
              className="card-img-top"
              src="./assets/heroes/marvel-daredevil.jpg"
            />
            <div
              className="card-body "
            >
              <p
                className="card-text fw-bold"
              >
                Matthew Michael Murdock
              </p>
              <p
                className="card-text"
              >
                Daredevil #1
              </p>
            </div>
            <div
              className="d-flex justify-content-between card-footer"
            >
              <small
                className="text-muted"
              >
                Marvel Comics
              </small>
              <a
                href="/heroe/marvel-daredevil"
                onClick={[Function]}
              >
                Más...
              </a>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
          >
            <div
              className="card-header"
            >
              <h5
                className="card-title fw-bolder"
              >
                Hawkeye
              </h5>
            </div>
            <img
              alt="Hawkeye"
              className="card-img-top"
              src="./assets/heroes/marvel-hawkeye.jpg"
            />
            <div
              className="card-body "
            >
              <p
                className="card-text fw-bold"
              >
                Clinton Francis Barton
              </p>
              <p
                className="card-text"
              >
                Tales of Suspense #57
              </p>
            </div>
            <div
              className="d-flex justify-content-between card-footer"
            >
              <small
                className="text-muted"
              >
                Marvel Comics
              </small>
              <a
                href="/heroe/marvel-hawkeye"
                onClick={[Function]}
              >
                Más...
              </a>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
          >
            <div
              className="card-header"
            >
              <h5
                className="card-title fw-bolder"
              >
                Cyclops
              </h5>
            </div>
            <img
              alt="Cyclops"
              className="card-img-top"
              src="./assets/heroes/marvel-cyclops.jpg"
            />
            <div
              className="card-body "
            >
              <p
                className="card-text fw-bold"
              >
                Scott Summers
              </p>
              <p
                className="card-text"
              >
                X-Men #1
              </p>
            </div>
            <div
              className="d-flex justify-content-between card-footer"
            >
              <small
                className="text-muted"
              >
                Marvel Comics
              </small>
              <a
                href="/heroe/marvel-cyclops"
                onClick={[Function]}
              >
                Más...
              </a>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
          >
            <div
              className="card-header"
            >
              <h5
                className="card-title fw-bolder"
              >
                Silver Surfer
              </h5>
            </div>
            <img
              alt="Silver Surfer"
              className="card-img-top"
              src="./assets/heroes/marvel-silver.jpg"
            />
            <div
              className="card-body "
            >
              <p
                className="card-text fw-bold"
              >
                Norrin Radd
              </p>
              <p
                className="card-text"
              >
                The Fantastic Four #48
              </p>
            </div>
            <div
              className="d-flex justify-content-between card-footer"
            >
              <small
                className="text-muted"
              >
                Marvel Comics
              </small>
              <a
                href="/heroe/marvel-silver"
                onClick={[Function]}
              >
                Más...
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
